plotly_express/_core.py

```diff
@@ -295,126 +287,141 @@ def configure_axes(args, constructor, fig, axes, orders):
+def set_cartesian_axis_opts(args, axis, letter, orders):
     log_key = "log_" + letter
     range_key = "range_" + letter
     if log_key in args and args[log_key]:
-        layout[axis]["type"] = "log"
+        axis["type"] = "log"
         if range_key in args and args[range_key]:
-            layout[axis]["range"] = [math.log(r, 10) for r in args[range_key]]
+            axis["range"] = [math.log(r, 10) for r in args[range_key]]
     elif range_key in args and args[range_key]:
-        layout[axis]["range"] = args[range_key]
+        axis["range"] = args[range_key]
 
     if args[letter] in orders:
-        layout[axis]["categoryorder"] = "array"
-        layout[axis]["categoryarray"] = (
+        axis["categoryorder"] = "array"
+        axis["categoryarray"] = (
             orders[args[letter]]
-            if axis.startswith("x")
+            if isinstance(axis, go.layout.XAxis)
             else list(reversed(orders[args[letter]]))
         )
 
 
-def configure_cartesian_marginal_axes(args, orders):
-    layout = dict()
+def configure_cartesian_marginal_axes(args, fig, orders):
+
     if "histogram" in [args["marginal_x"], args["marginal_y"]]:
-        layout["barmode"] = "overlay"
-    for letter in ["x", "y"]:
-        layout[letter + "axis1"] = dict(
-            title=get_decorated_label(args, args[letter], letter)
-        )
-        set_cartesian_axis_opts(args, layout, letter, letter + "axis1", orders)
-    for letter in ["x", "y"]:
-        otherletter = "x" if letter == "y" else "y"
-        if args["marginal_" + letter]:
-            if args["marginal_" + letter] == "histogram" or (
-                "color" in args and args["color"]
-            ):
-                main_size = 0.74
-            else:
-                main_size = 0.84
-            layout[otherletter + "axis1"]["domain"] = [0, main_size]
-            layout[otherletter + "axis1"]["showgrid"] = True
-            layout[otherletter + "axis2"] = {
-                "domain": [main_size + 0.005, 1],
-                "showticklabels": False,
-            }
-    return layout
+        fig.layout["barmode"] = "overlay"
+
+    nrows = len(fig._grid_ref)
+    row_step = 2 if args["marginal_x"] else 1
+
+    ncols = len(fig._grid_ref[0])
+    col_step = 2 if args['marginal_y'] else 1
+
+    # Set y-axis titles and axis options in the left-most column
+    for yaxis in fig.select_yaxes(col=1):
+        set_cartesian_axis_opts(args, yaxis, 'y', orders)
 
+    # Set x-axis titles and axis options in the bottom-most row
+    for xaxis in fig.select_xaxes(row=1):
+        set_cartesian_axis_opts(args, xaxis, 'x', orders)
+
+    # Configure axis ticks on marginal subplots
+    if args['marginal_x']:
+        for row in range(2, nrows+1, row_step):
+            fig.update_yaxes(
+                showticklabels=False,
+                showgrid=args["marginal_x"] == 'histogram',
+                row=row
+            )
+            fig.update_xaxes(
```

Comment:
has this been run through `black` ? :)

Reply:
Done now

Reply:
Thanks! I know the plotly.py repo doesn't have that in place but this one does for now ;)

Reply:
Yeah, I'm interested in using black in plotly.py too.  Have you looked into the git integration https://black.readthedocs.io/en/stable/version_control_integration.html?